core/src/main/scala/com/microsoft/azure/synapse/ml/codegen/DotnetWrappable.scala

```diff
@@ -0,0 +1,519 @@
+      case _: ArrayMapParam =>
+        s"""
+           |$docString
+           |public ${getParamInfo(p).dotnetType} Get$capName()
+           |{
+           |    JvmObjectReference[] jvmObjects = (JvmObjectReference[])Reference.Invoke(\"get$capName\");
+           |    Dictionary<string, object>[] result = new Dictionary<string, object>[jvmObjects.Length];
+           |    JvmObjectReference hashMap;
+           |    JvmObjectReference[] keySet;
+           |    Dictionary<string, object> dic;
+           |    object val;
+           |    for (int i = 0; i < result.Length; i++)
+           |    {
+           |        hashMap = (JvmObjectReference)SparkEnvironment.JvmBridge.CallStaticJavaMethod(
+           |            "com.microsoft.azure.synapse.ml.codegen.DotnetHelper", "convertToJavaMap", jvmObjects[i]);
+           |        keySet = (JvmObjectReference[])(
+           |            (JvmObjectReference)hashMap.Invoke("keySet")).Invoke("toArray");
+           |        dic = new Dictionary<string, object>();
+           |        foreach (var k in keySet)
+           |        {
+           |            val = SparkEnvironment.JvmBridge.CallStaticJavaMethod(
+           |                "com.microsoft.azure.synapse.ml.codegen.DotnetHelper",
+           |                "mapScalaToJava", hashMap.Invoke("get", k));
+           |            dic.Add((string)k.Invoke("toString"), val);
+           |        }
+           |        result[i] = dic;
+           |    }
+           |    return result;
+           |}
+           |""".stripMargin
+      case _: EstimatorParam | _: ModelParam | _: TransformerParam | _: EvaluatorParam | _: PipelineStageParam =>
+        s"""
+           |$docString
+           |public ${p.asInstanceOf[WrappableParam[_]].dotnetReturnType} Get$capName()
+           |{
+           |    JvmObjectReference jvmObject = (JvmObjectReference)Reference.Invoke(\"get$capName\");
+           |    var (constructorClass, methodName) = Helper.GetUnderlyingType(jvmObject);
+           |    Type type = Type.GetType(constructorClass);
+           |    MethodInfo method = type.GetMethod(methodName, BindingFlags.NonPublic | BindingFlags.Static);
+           |    return (${p.asInstanceOf[WrappableParam[_]].dotnetReturnType})method.Invoke(
+           |        null, new object[] {jvmObject});
+           |}
+           |""".stripMargin
+      case _: ArrayParamMapParam =>
+        s"""
+           |$docString
+           |public ParamMap[] Get$capName()
+           |{
+           |    JvmObjectReference[] jvmObjects = (JvmObjectReference[])Reference.Invoke(\"get$capName\");
+           |    ParamMap[] result = new ParamMap[jvmObjects.Length];
+           |    for (int i=0; i < jvmObjects.Length; i++)
+           |    {
+           |        result[i] = new ParamMap(jvmObjects[i]);
+           |    }
+           |    return result;
+           |}
+           |""".stripMargin
+      case _: TransformerArrayParam | _: EstimatorArrayParam =>
```

Comment:
likewise here, in the future perhaps this is the content that we try to push into the class def of a new param class